ui/API_scripts.js

```diff
@@ -1,119 +1,120 @@
+// eslint-disable-next-line no-unused-vars
 const prevEPRoll = function() {
-    var total = 0;
-    var text = '';
-    var any_positive = false;
-    var any_negative = false;
-    var must_roll = true;
-    while (must_roll) {
-	must_roll = false;
-	const only_positive = any_positive && !any_negative
-	const only_negative = !any_positive && any_negative
-	for (var i = 0; i < 3; i++) {
-	    const die = randomInteger(6);
-	    var roll;
-	    if (die === 6) {
-		must_roll = true
-	        roll = "\u29C9"; // two joined squares character
-	    } else {
-		var roll = die - 3
-		if (only_positive) {
-		    roll = Math.abs(roll);
-		} else if (only_negative) {
-		    roll = -Math.abs(roll);
-		}
-                total = total + roll;
-	        if (roll < 0) {
-		    any_negative = true;
-		    roll = '<span style="color:red";>' + roll + '</span>'
-	        } else if (roll >  0) {
-		    any_positive = true;
-		    roll = '<span style="color:blue";>+' + roll + '</span>'
-	        }
-	    }
-	    if (text === '') {
-	        text = String(roll)
-	    } else {
-                text = text + ", " + roll;
-	    }
-	}
+  let total = 0;
+  let text = '';
+  let any_positive = false;
+  let any_negative = false;
+  let must_roll = true;
+  while (must_roll) {
+    must_roll = false;
+    const only_positive = any_positive && !any_negative;
+    const only_negative = !any_positive && any_negative;
+    for (let i = 0; i < 3; i++) {
+      const die = randomInteger(6);
+      let roll;
+      if (die === 6) {
+        must_roll = true;
+        roll = '\u29C9'; // two joined squares character
+      } else {
+        roll = die - 3;
```

Comment:
This inner `var` was in error. All `var`'s are hoisted to the root of the function, which makes this sort of override line 13, but in an ambiguous way.
The intention of the code here is to redefine `roll`, so simply removed `var` here.